akka-actor/src/main/scala/akka/actor/dungeon/ReceiveTimeout.scala

```diff
@@ -60,12 +60,12 @@ private[akka] trait ReceiveTimeout { this: ActorCell =>
 
   protected def cancelReceiveTimeoutIfNeeded(message: Any): (Duration, Cancellable) = {
     if (hasTimeoutData && !message.isInstanceOf[NotInfluenceReceiveTimeout])
-      cancelReceiveTimeout()
+      cancelReceiveTimeoutTask()
 
     receiveTimeoutData
   }
 
-  override final def cancelReceiveTimeout(): Unit =
+  private def cancelReceiveTimeoutTask(): Unit =
```

Comment:
Wouldn't it be more clear we rather remove the method implementation in `akka.actor.AbstractActor.ActorContext#cancelReceiveTimeout` and make that abstract (as that currently has a different logic now, leaving the cancellable around)?